lib/client-api/src/story_store.js

```diff
@@ -134,28 +134,48 @@ export default class StoryStore extends EventEmitter {
     delete _data[id];
   };
 
-  addStory({ id, kind, name: storyName, story, getDecorated, parameters = {} }) {
+  addStory(
+    { id, kind, name, storyFn: original, parameters = {} },
+    { getDecorators, applyDecorators }
+  ) {
     const { _data } = this;
 
     if (_data[id]) {
       logger.warn(stripIndents`
-      Story with id ${id} already exists in the store!
+        Story with id ${id} already exists in the store!
 
-      Perhaps you added the same story twice, or you have a name collision?
-      Story ids need to be unique -- ensure you aren't using the same names modolo url-sanitization.`);
+        Perhaps you added the same story twice, or you have a name collision?
+        Story ids need to be unique -- ensure you aren't using the same names modolo url-sanitization.
+      `);
     }
 
-    _data[id] = toChild({
+    const identification = {
+      id,
       kind,
-      name: storyName,
-      story,
+      name,
+      story: name, // legacy
+    };
+
+    // immutable original storyFn
+    const getOriginal = memoize(1)(() => original);
```

Comment:
What is the point of memoizing a function that returns a reference? Doesn't this just waste memory?

Reply:
true I guess